src/click/core.py

```diff
@@ -2011,7 +2011,15 @@ def _convert(value, level):
             if level == 0:
                 return self.type(value, self, ctx)
 
-            return tuple(_convert(x, level - 1) for x in value)
+            try:
+                iter_value = iter(value)
+            except TypeError:
+                raise TypeError(
+                    "Default for parameter with multiple = True or nargs > 1"
```

Comment:
This method is called for all values, not only default values.

Reply:
I'll change the error message